servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilderTest.java

```diff
@@ -222,6 +223,45 @@ void internalClientsUseDifferentExecutionContextWhenConfigured() throws Exceptio
         }
     }
 
+    @Test
+    void internalClientSchemeShouldBeCaseInsensitive() throws Exception {
+        HttpExecutionContext expectedCtx = new HttpExecutionContextBuilder()
+                .ioExecutor(CTX.ioExecutor())
+                .executor(CTX.executor())
+                .bufferAllocator(PREFER_DIRECT_ALLOCATOR)
+                .executionStrategy(offloadNone())
+                .build();
+        AtomicInteger counter = new AtomicInteger();
+
+        try (ServerContext serverContext = HttpServers.forAddress(localAddress(0))
+                .executionStrategy(offloadNone())
+                .listenStreamingAndAwait((ctx, request, responseFactory) -> succeeded(responseFactory.ok()));
+             BlockingHttpClient blockingHttpClient = HttpClients.forMultiAddressUrl(getClass().getSimpleName())
+                     .initializer((scheme, address, builder) ->
+                             builder.appendClientFilter(
+                                     client -> {
+                                         counter.incrementAndGet();
+                                         return new StreamingHttpClientFilter(client) {
+                                         };
+                                     }))
+                     .ioExecutor(expectedCtx.ioExecutor())
+                     .executor(expectedCtx.executor())
+                     .bufferAllocator(expectedCtx.bufferAllocator())
+                     .executionStrategy(expectedCtx.executionStrategy())
+                     .buildBlocking()) {
+
+            HttpResponse response = blockingHttpClient.request(
+                    blockingHttpClient.get("http://" + serverHostAndPort(serverContext)));
+            assertThat(response.status(), is(OK));
+
+            response = blockingHttpClient.request(
+                    blockingHttpClient.get("HTTP://" + serverHostAndPort(serverContext)));
+            assertThat(response.status(), is(OK));
+
+            assertThat(counter.get(), is(1));
```

Comment:
Thanks for the reproducer! An alternative (minimal) way to fix the issue without introducing new types will be:
```patch
Index: servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java b/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java
--- a/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java	(revision f4d7153d5606531fbdd6640d29502970b19e8c97)
+++ b/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java	(date 1716514112838)
@@ -58,6 +58,7 @@
 
 import java.net.InetSocketAddress;
 import java.net.MalformedURLException;
+import java.util.Locale;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.function.Function;
@@ -165,15 +166,16 @@
                                 ", expected absolute-form URL");
             }
 
-            final String scheme = metaData.scheme();
-            if (scheme == null) {
+            final String originalScheme = metaData.scheme();
+            if (originalScheme == null) {
                 throw new MalformedURLException("Request-target does not contains scheme: " +
                         metaData.requestTarget() + ", expected absolute-form URL");
             }
+            final String scheme = originalScheme.toLowerCase(Locale.ENGLISH);
 
             final int parsedPort = metaData.port();
             final int port = parsedPort >= 0 ? parsedPort :
-                    (HTTPS_SCHEME.equalsIgnoreCase(scheme) ? defaultHttpsPort : defaultHttpPort);
+                    (HTTPS_SCHEME.equals(scheme) ? defaultHttpsPort : defaultHttpPort);
             setHostHeader(metaData);
             metaData.requestTarget(absoluteToRelativeFormRequestTarget(metaData.requestTarget(), scheme, host));
```
Another option is to always lowercase inside `DefaultHttpRequestMetaData.scheme()` implementation to aligs with its javadoc. 

To reduce memory allocations in both cases, we can enhance the code to guess the standard schemes first before doing lower casing.

Reply:
After thinking about it more, it feels like the most minimalistic and efficient way is to enhance this line:
https://github.com/apple/servicetalk/blob/b8149d36ba1d4cddf4b034b18f3cdb4c5a358adc/servicetalk-http-api/src/main/java/io/servicetalk/http/api/Uri3986.java#L172

We can have an internal static function that will return a lowercase scheme. For well-known and expected schemes (http and https) it can even avoid allocation of a new string by checking the first 4-5 chars of `uri` and returning a corresponding constant. 

WDYT?

Reply:
Thank you for your kind review, @idelpivnitskiy!

I also prefer the second one and apply it. It becomes much more minimal.

nit: `Locale.ROOT` instead of `Locale.ENGLISH`?

> To obtain correct results for locale insensitive strings, use {@code toLowerCase(Locale.ROOT)}.

Reply:
idk, feels more predictable on any OS. based on IANA list of registered schemes, all of them are in English: https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml

Reply:
Thank you, I see.